crates/services/producer/src/block_producer.rs

```diff
@@ -216,14 +221,23 @@ where
         )
         .await?;
 
-        for (transaction, tx_status) in transactions.iter().zip(tx_statuses.iter()) {
-            let is_script = transaction.is_script();
-            if is_script && tx_status.receipts.is_empty() {
-                return Err(anyhow!("Expected at least one set of receipts"))
-            }
+        if are_txs_script
+            .iter()
+            .zip(tx_statuses.iter())
+            .any(|(is_script, tx_status)| *is_script && tx_status.receipts.is_empty())
+        {
+            return Err(anyhow!("Expected at least one set of receipts"));
```

Comment:
You don't need to `return` here, you can just have the else case be `Ok`.

Reply:
interesting in other languages I was taught it is preferable to eliminate the else since it is unnecessary.  So in rust it is preferable to eliminate return statements instead of the else case?

Reply:
Its preference. I just don't like returns. The compiled code won't be very different.